File: whatscooking/src/main/java/com/rll/whatscooking/domain/Nutrition.java
package com.rll.whatscooking.domain;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Data;

@Entity
@Data
public class Nutrition {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    private Integer calories;
    private Double protein;
    private Double fat;
    private Double carbs;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "nutritionist_id")
    private User user;

}

File: Client/src/app/Models/Nutrition.ts
import { User } from './User';

export class Nutrition {
  nutrition_id?: number;
  calories?: number;
  protein?: number;
  fat?: number;
  carbs?: number;
  user?: User;

  constructor(
    nutrition_id?: number,
    calories?: number,
    protein?: number,
    fat?: number,
    carbs?: number,
    user?: User
  ) {
    this.nutrition_id = nutrition_id;
    this.calories = calories;
    this.protein = protein;
    this.fat = fat;
    this.carbs = carbs;
    this.user = user;
  }
}
